File: evidence/generated-systems/golang/hr-on-golang/hr-on-golang/internal/model/PerformanceReview.go
package model

import (
    "time"
    "gorm.io/gorm"
)

//==============================================================
// PerformanceReview Declaration
//==============================================================
type PerformanceReview struct {
    gorm.Model
     ReviewNumber                                    string
    ReviewDate                                                            time.Time
    ReviewerComments                                    string
    EmployeeId         *uint
    Employee           *Employee `gorm:"foreignKey:EmployeeId"`
    ReviewerId         *uint
    Reviewer           *Employee `gorm:"foreignKey:ReviewerId"`
    CycleId         *uint
    Cycle           *PerformanceCycle `gorm:"foreignKey:CycleId"`
     CompetencyRatings           []CompetencyRating `gorm:"foreignKey:CompetencyRatingsFromPerformanceReviewId"`
     Goals           []Goal `gorm:"foreignKey:GoalsFromPerformanceReviewId"`
    Rating                      PerformanceRating
    Status                      ReviewStatus

// parent associations as their child

}

File: evidence/generated-systems/rubyonrails/hr-on-rails/hronrails/app/models/performancereview.rb
class PerformanceReview < ApplicationRecord
  enum Rating: [:Unsatisfactory, :NeedsImprovement, :MeetsExpectations, :ExceedsExpectations, :Outstanding]
  enum Status: [:NotStarted, :InProgress, :Finalized, :Acknowledged]


  has_many :Employee, class_name: 'Employee'
  has_many :Reviewer, class_name: 'Employee'
  has_many :Cycle, class_name: 'PerformanceCycle'
  has_many :CompetencyRatings, class_name: 'CompetencyRating'
  has_many :Goals, class_name: 'Goal'

end
